### PR TITLE
u3-tool: 0.3 -> 1.0

### PR DESCRIPTION
**Pull request by makefu**
Update u3-tool to latest version (2009 -> 2012).
Also move tool to by-name package structure.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/u3/u3-tool/package.nix [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
   pname = "u3-tool";
-  version = "0.3";
+  version = "1.0";
 
   enableParallelBuilding = true;
 
   src = fetchurl {
-    url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.gz";
+    url = "mirror://sourceforge/${finalAttrs.pname}/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
     sha256 = "1p9c9kibd1pdbdfa0nd0i3n7bvzi3xg0chm38jg3xfl8gsn0390f";
```
Comment (FliegendeWurst) [inline pkgs/by-name/u3/u3-tool/package.nix:11]: All mirrors return 404 for this URL

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/u3/u3-tool/package.nix (hunk 1 above)
Comment (makefu) [reply pkgs/by-name/u3/u3-tool/package.nix:11]: you are right, i fogor to update checksum which resulted in a cache hit :)

File: pkgs/by-name/u3/u3-tool/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    autoreconfHook
+  ];
+
+
+  meta = with lib; {
+    description = "Tool for controlling the special features of a 'U3 smart drive' USB Flash disk";
+    homepage = "https://sourceforge.net/projects/u3-tool/";
+    license = licenses.gpl2Plus;
+    platforms = with platforms; linux;
+    maintainers = with maintainers; [ makefu ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/u3/u3-tool/package.nix:32]:
```suggestion
  meta = {
    description = "Tool for controlling the special features of a 'U3 smart drive' USB Flash disk";
    homepage = "https://sourceforge.net/projects/u3-tool/";
    license = lib.licenses.gpl2Plus;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ makefu ];
```

File: pkgs/by-name/u3/u3-tool/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv
+, fetchFromGitHub
```
Comment (FliegendeWurst) [inline pkgs/by-name/u3/u3-tool/package.nix:2]: Please format with the new formatter.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Review by donovanglover — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**